### DISCUSSION TITLE
Advanced debuging options for "Ensure that the Magewire component wasn't tampered with"

### DISCUSSION THREAD

**Discussion by Aquive** [magewirephp/magewire#150] [Ideas]
When working with Magewire I get this error very frequently:

`{"message":"Magewire security vulnerability: Magewire encountered corrupt data\n when trying to hydrate the Hyva\\ColorOptions\\Magewire\\ColorPicker component. Ensure that the [name, id, resolver and data] of the Magewire component wasn't\n tampered with between requests.","code":500}`

It would be awesome to have a debug function which logs the original and "tampered" one so it is easy to spot the difference. When encoutering this issue it is hard to spot whats going on right now.

**Comment by lzyenjoy** [comment]
Hyva\\ColorOptions\\Magewire\\ColorPicker
Make sure the properties inside are protected or private

Eg: protected Data $helper;
